### PR TITLE
fix: Microsoft.OpenAPI package reference conflict

### LINKED ISSUE #1717: BUG: Application is incompatible with Microsoft.OpenAPI 1.2.0, but this package version is allowed in package config.
Package Swashbuckle.AspNetCore.Swagger - latest version

Steps to reproduce:
- Setup base project with Swagger Connected and swaggergen enabled
- Add API endpoint that contains default parameter
- Add Microsoft.OpenAPI package version 1.2.0 to project
- Try to access swagger.json

Result: Internal Error appear. Stack Trace details:
```
System.MissingMethodException: Method not found: 'Void Microsoft.OpenApi.Any.OpenApiString..ctor(System.String)'.
   at Swashbuckle.AspNetCore.SwaggerGen.OpenApiAnyFactory.CreateFor(OpenApiSchema schema, Object value)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.ApplyParameterMetadata(OpenApiSchema schema, Type type, ParameterInfo parameterInfo)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaGenerator.GenerateSchema(Type type, SchemaRepository schemaRepository, MemberInfo memberInfo, ParameterInfo parameterInfo)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateParameter(ApiParameterDescription apiParameter, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.<>c__DisplayClass10_0.<GenerateParameters>b__1(ApiParameterDescription apiParam)
   at System.Linq.Enumerable.WhereSelectListIterator`2.ToList()
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateParameters(ApiDescription apiDescription, SchemaRepository schemaRespository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperation(ApiDescription apiDescription, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GenerateOperations(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GeneratePaths(IEnumerable`1 apiDescriptions, SchemaRepository schemaRepository)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GetSwagger(String documentName, String host, String basePath)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
```

The issue is because Microsoft.OpenAPI v 1.2.0 contains no constructor for `OpenApiString` class with only string input parameter.

Fix suggestion: Restrict max Microsoft.OpenAPI version in Swashbuckle.AspNetCore.Swagger project not to include v 1.2.0

### PR DESCRIPTION
**Pull request by korifeich**
Add upper bound to Miicrosoft.OpenAPI package reference to Swashbuckle.AspNetCore.Swagger project to be less than 1.2.0 as it breaks contains breaking changes for  Swashbuckle.AspNetCore.Swagger project

fixes #1717

### CODE REVIEW

**Comment by CumpsD** [comment]
Better to put some pressure on them to fix it properly so everybody can move to 1.2.x (or 1.3.0 if they want to name it that)

https://github.com/microsoft/OpenAPI.NET/issues/487#issuecomment-648976686

**Comment by CumpsD** [comment]
Wait to merge this, fixing it at the source: https://github.com/microsoft/OpenAPI.NET/pull/492

File: src/Swashbuckle.AspNetCore.Swagger/Swashbuckle.AspNetCore.Swagger.csproj [hunk 1]
```diff
@@ -19,7 +19,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.OpenApi" Version="1.1.4" />
+    <PackageReference Include="Microsoft.OpenApi" Version="[1.1.4,1.2.0)" />
```
Comment (CumpsD) [inline src/Swashbuckle.AspNetCore.Swagger/Swashbuckle.AspNetCore.Swagger.csproj:22]:
```suggestion
    <PackageReference Include="Microsoft.OpenApi" Version="1.2.2" />
```

We fixed it at the source, Microsoft released a version without the breaking changes. Swashbuckle can now target 1.2.2

**Review by RFlipper — CHANGES_REQUESTED** [review]
The PR is outdated and should be rejected

**Pull request closed without merging**